application/controllers/artifacts_controller.py

```diff
@@ -83,6 +85,25 @@ def delete(self, params, object_id):
         except NotFound:
             return {"error": "not found"}, 404
 
+    @staticmethod
+    def save_sizes(filename):
+        try:
+            image = Image.open(filename)
+        except IOError:
+            return
+
+        [prefix, suffix] = filename.rsplit('.', 1)
+        widths = [320, 480, 640, 750, 1080]
+
+        for width in widths:
+            width_percent = (width / float(image.size[0]))
+            height = int((float(image.size[1]) * float(width_percent)))
+            resized_image = image.resize((width, height), Image.ANTIALIAS)
+
+            new_filename = f"{prefix}_{width}w.{suffix}"
+            resized_image.save(new_filename)
```

Comment:
You know exactly what I am going to say about this...
- too long  
- doesn't have anything to do with the controller
- should be a seperate class

For inspiration check my PR where i refactored the image upload. You might even wanna merge although there is a lot of different stuff in one PR again. Sorry about that!

Reply:
Dat PR advertisement 😂

Reply:
Please check my updates.